### PR TITLE
Various editorial corrections

### PR DESCRIPTION
**Pull request by LewisJEllis**
9 fixes; most are pretty cut and dry, but three less obvious ones are:
* the will/would conditional
* ~~function vs method~~
* provide vs provides

I do think the changes here are preferable, but the prior forms weren't *necessarily* incorrect.

I also noted 9 or 10 other corrections/problems while reading, but it looks like @ShMcK fixed most of those in his recent commits, so props to him.

### CODE REVIEW

File: manuscript/11-Promises.md [hunk 1]
```diff
@@ -71,31 +71,31 @@ In this code, a successful call to `readFile()` results in another asynchronous
 While this works fairly well, you can quickly get into a pattern that has come to be known as *callback hell*. Callback hell occurs when you nest too many callbacks:
 
 ```js
-method1(function(err, result) {
+function1(function(err, result) {
```
Comment (nzakas) [inline manuscript/11-Promises.md]: Not a fan of this change. It's visually hard to parse. I get the point, that these aren't methods, but I don't like this naming.

Comment (LewisJEllis) [inline manuscript/11-Promises.md]: Fair enough, pushed new version.

**Comment by nzakas** [comment]
Thanks!
